### DISCUSSION TITLE
Clarification on the support of process substitution in OpenSSL

### DISCUSSION THREAD

**Discussion by hrafnthor** [openssl/openssl#21456] [Q&A] [answered]
I have be unable to use process substitution with OpenSSL when doing so via `bash` while, on the same system, having no issues running the same command with `zsh`.

The command in question is the following (with a valid credential replacing mock value):

```shell
echo "something to sign" | openssl dgst -binary -sha256 -sign <(echo "credential")
```
When run via `bash` (verified via `ps -p $$`) the command will fail with error:

```shell
Could not read private key from /dev/fd/63
``` 

Again I stress, the same command above works flawlessly when run using `zsh`.

Now the question becomes can this behaviour be attributed to OpenSSL it self, or would this be due to some quirk in the implementations of the shells? 

I ask since there are numerous examples online showing OpenSSL being used with process substitution, indicating that for some this clearly does work when used with `bash`, but for others it does not work at all (no matter what the shell is).

---

#### System information
Ubuntu 22.04 - 5.19.0-46-generic
OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)
zsh 5.8.1 (x86_64-ubuntu-linux-gnu)
GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)

**Comment by hrafnthor** [comment, marked as answer]
It feels very silly now but it turns out that the differences in behaviour were due to backslashes in the credential payload, and `bash` requiring interpretation of backslash escapes to be turned on via the -e flag (i.e using `... <(echo -e "credentials")`.

**Reply by mspncp** [reply]
It's not silly at all. Instead, it should feel satisfying that you were able to solve your riddle all by your own. :-)
